### PR TITLE
[CELEBORN-822][DOC] Introduce a quick start guide for running Apache Flink with Apache Celeborn

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Introduce a quick start guide for running Apache Flink with Apache Celeborn to help Flink users to run with Celeborn.

### Why are the changes needed?

There is no quick start guide for running Apache Flink with Apache Celeborn.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

None.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ cd $CELEBORN_HOME
 ```
 You should see `Master`'s ip:port in the log:
 ```log
-INFO [main] NettyRpcEnvFactory: Starting RPC Server [MasterSys] on 192.168.2.109:9097
+INFO [main] NettyRpcEnvFactory: Starting RPC Server [MasterSys] on 192.168.2.109:9097 with advisor endpoint 192.168.2.109:909
```
Comment (RexXiong) [inline docs/README.md:54]: nit: with advisor endpoint 192.168.2.109:909 -> with advisor endpoint 192.168.2.109:9097

File: docs/README.md [hunk 2]
```diff
@@ -115,3 +116,36 @@ INFO [dispatcher-event-loop-9] Controller: Reserved 10 primary location and 0 re
 INFO [dispatcher-event-loop-8] Controller: Start commitFiles for local-1690000152711-0
 INFO [async-reply] Controller: CommitFiles for local-1690000152711-0 success with 10 committed primary partitions, 0 empty primary partitions, 0 failed primary partitions, 0 committed replica partitions, 0 empty replica partitions, 0 failed replica partitions.
 ```
+
+## Start Flink with Celeborn
+#### Copy Celeborn Client to Flink's lib
+Celeborn release binary contains clients for Flink 1.14.x, Flink 1.15.x and Flink 1.17.x, copy the corresponding client jar into Flink's
+`jar/` directory:
```
Comment (RexXiong) [inline docs/README.md:123]: let's use lib/ directory

File: docs/README.md [hunk 3]
```diff
@@ -115,3 +116,36 @@ INFO [dispatcher-event-loop-9] Controller: Reserved 10 primary location and 0 re
...
+vi conf/flink-conf.yaml
+```
+```properties
+shuffle-service-factory.class: org.apache.celeborn.plugin.flink.RemoteShuffleServiceFactory
+```
+Then deploy the example word count job to the running cluster:
+```shell
+cd $FLINK_HOME
+
+./bin/flink run examples/streaming/WordCount.jar
```
Comment (RexXiong) [inline docs/README.md:140]: Need explicit set execution-mode of the WordCount job as BATCH

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main

**Pull request closed without merging**